torch/distributions/gamma.py

```diff
@@ -27,14 +27,18 @@ class Gamma(ExponentialFamily):
             (often referred to as beta)
     """
     arg_constraints = {'concentration': constraints.positive, 'rate': constraints.positive}
-    support = constraints.positive
+    support = constraints.nonnegative
     has_rsample = True
     _mean_carrier_measure = 0
 
     @property
     def mean(self):
         return self.concentration / self.rate
 
+    @property
+    def mode(self):
+        return ((self.concentration - 1) / self.rate).clamp(0)
```

Comment:
We might want to `.clamp(min=finfo(...).tiny)` and keep the `positive` constraint.

nit: it's a bit clearer to write `.clamp(min=0)`

Reply:
I've added some tests in https://github.com/pytorch/pytorch/pull/76690/commits/8fa7baded13c90588191218754315dda3f8432e9.